src/models/smtp.js

```diff
@@ -1,41 +1,78 @@
+import _ from 'lodash';
 export default function(sequelize, DataTypes) {
-	const smtp = sequelize.define("SMTP", {
-		email: {
-			type: DataTypes.STRING,
-			unique: true,
-		},
-		password: DataTypes.STRING,
-		smtp_server: DataTypes.STRING,
-		server_port: DataTypes.INTEGER,
-		type: {
-			type: DataTypes.ENUM,
-			values: ["SSL", "TLS"],
-		},
-		status: {
-			type: DataTypes.ENUM,
-			values: ["TRUE", "FALSE"],
-			defaultValue: "FALSE",
-		},
-	}, {
-		timestamps: true,
-		freezeTableName: true,
-		allowNull: true,
-		hooks: {
+    const smtp = sequelize.define("SMTP", {
+        email: {
+            type: DataTypes.STRING,
+            unique: true,
+        },
+        password: DataTypes.STRING,
+        smtp_server: DataTypes.STRING,
+        server_port: DataTypes.INTEGER,
+        type: {
+            type: DataTypes.ENUM,
+            values: ["SSL", "TLS"],
+        },
+        status: {
+            type: DataTypes.ENUM,
+            values: ["TRUE", "FALSE"],
+            defaultValue: "FALSE",
+        },
+    }, {
+        timestamps: true,
+        freezeTableName: true,
+        allowNull: true,
+        hooks: {
 
-			beforeCreate: function(SMTP) {
-				return new Promise((resolve, reject) => {
-					this.findOne({ where: { email: SMTP.email } })
+            beforeCreate: function(SMTP) {
+                return new Promise((resolve, reject) => {
+                    this.findOne({ where: { email: SMTP.email } })
                         .then((email) => {
-	if (email) {
-		reject("Email Already In Use");
-	} else {
-		resolve();
-	}
-});
-				});
+                            if (email) {
+                                reject("Email Already In Use");
+                            } else {
+                                resolve();
+                            }
+                        });
+                });
 
-			}
-		}
-	});
-	return smtp;
+            }
+        },
+        classMethods: {
+            smtpTest(email) {
+                return new Promise((resolve, reject) => {
+                    this.findOne({ where: { email: email } })
+                        .then((data) => {
+                            if (data) {
+                                this.findAll({})
+                                    .then((data) => {
+                                        _.map(data, (val, key) => {
+                                            if (val.email == email) {
+                                                this.update({ status: "TRUE" }, { where: { email: email } })
```

Comment:
why are we updating status in smtpTest function?

and why are we doing findAll({}) all these. 

and why are not using boolean for True, false

find a better logic and better code.